cou/steps/plan.py

```diff
@@ -491,48 +524,65 @@ def _get_archive_data_steps(analysis_result: Analysis, args: CLIargs) -> list[Pr
     return []
 
 
-def _get_post_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[PostUpgradeStep]:
-    """Get the post-upgrade steps.
+def _get_set_noout_steps(analysis_result: Analysis, args: CLIargs) -> list[PreUpgradeStep]:
+    """Get set noout steps.
 
     :param analysis_result: Analysis result
     :type analysis_result: Analysis
     :param args: CLI arguments
     :type args: CLIargs
-    :return: List of post-upgrade steps.
+    :return: List of pre-upgrade steps.
     :rtype: list[PreUpgradeStep]
     """
-    steps = []
-    if args.upgrade_group in {DATA_PLANE, None}:
-        steps.extend(_get_ceph_mon_post_upgrade_steps(analysis_result.apps_control_plane))
-
-    return steps
+    if args.set_noout and args.upgrade_group in {DATA_PLANE, None}:
```

Comment:
Here I think there is another problem.

from the chart in our channel the `noout` should be `unset` before upgrading. So if the cloud has it set and the user runs with the flag is not halting, it will simply set `noout` without checking the cloud.

In another words:
I think the `assert_osd_noout_state` should always happen to ensure that is unset.

Reply:
but this will also has weird user experience: if the operator has the habit to always set noout before upgrading (e.g. `cou upgrade`), then `cou` will always blocks the operator until he change his habit. 

The idea is that we don't check if `noout` unset, when the user add `--set-noout` because they are expecting `noout` is set anyways

Reply:
I think this is will be more appropriate when we think of separating the pre upgrade sanity check and post upgrade sanity check out of the upgrade plan (the big box of step). 

Separating the sanity checks will allow us to do easy-to-read & comprehensive sanity check report to instruct the operator how to make it work with COU (before and after upgrade), and do not clutter the stdout with unnecessary upgrade plan when pre upgrade sanity check failed (like a `cou lint`)

Reply:
Anyways, reverted back to my original design (i.e. the one you mentioned) as it was requested by user